tests/macos/xcconfig/tests.bzl

```diff
@@ -177,11 +182,15 @@ def xcconfig_unit_test_suite():
                 xcconfig = {"GCC_PREPROCESSOR_DEFINITIONS": ["DISPLAY_VERSION=1.0.0-beta.1", "SDK_NAME=WHY WOULD YOU ADD SPACES"]},
                 expected = {"objc_copts": ["-D'DISPLAY_VERSION=1.0.0-beta.1'", "-D'SDK_NAME=WHY WOULD YOU ADD SPACES'"]},
             ),
-            # TODO: we should eventually support conditioned vars somehow
             assert_xcconfig(
-                name = "conditioned_option_unsupported",
+                name = "conditioned_option",
                 xcconfig = {"CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER": "YES"},
-                error = 'CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER: option specifies a condition ("$(CLANG_UNDEFINED_BEHAVIOR_SANITIZER) == YES"), which is currently unsupported',
+                expected = {},
```

Comment:
Why isn't the expected value ```{"objc_copts": ["-fsanitize=integer"]}```

Reply:
I think there might be a bug in xcspecs_eval.bzl. It should say:

```
def _com_apple_compilers_llvm_clang_1_0__CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER__Condition(xcconfigs, id_configs):
    # $(CLANG_UNDEFINED_BEHAVIOR_SANITIZER) == YES

    used_user_content = False
    
    eval_val_0 = ""
   # bug: currently, eval_key_0 is "CLANG_UNDEFINED_BEHAVIOR_SANITIZER", but the key passed to xcconfig is "CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER"
    eval_key_0 = "CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER"
```

Reply:
why is this a bug? The actual condition for the option is `$(CLANG_UNDEFINED_BEHAVIOR_SANITIZER) == YES`, which isn't set here

Reply:
OK, so you're saying that setting the `CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER` key in xcconfig is a no-op. Can we rename it to something else in this test ... like:
```
                xcconfig = {"foobar": "YES"},
                expected = {},
```

To someone reading this test case, the `CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER` looks just like the `CLANG_UNDEFINED_BEHAVIOR_SANITIZER` key, which would actually result in the "objc_copts" getting set to ["-fsanitize=integer"].

Reply:
but `foobar` isn't a valid xcconfig.... you can look in the xcspec bzl file to find all the one's that we're operating over

Reply:
All right. Maybe there should be a comment here ... because when I read this test case at first, I would have assumed that setting `CLANG_UNDEFINED_BEHAVIOR_SANITIZER_INTEGER` to YES would change the copts ... when in fact, it is only setting `CLANG_UNDEFINED_BEHAVIOR_SANITIZER` to YES that would affect cops.